src/main/groovy/wooga/gradle/github/base/GithubBasePlugin.groovy

```diff
@@ -43,54 +44,44 @@ class GithubBasePlugin implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
+        Grgit git = Grgit.init(dir: project.rootProject.projectDir)
+        RepositoryInfo repoInfo = new RepositoryInfo(project, git)
+
         GithubPluginExtension extension = project.extensions.create(EXTENSION_NAME, DefaultGithubPluginExtension.class, project)
 
         extension.username.set(project.provider({
-            def value = project.properties.get(GithubBasePluginConvention.GITHUB_USER_NAME_OPTION)
-            if (value) {
-                return value.toString()
-            }
-            null
+            project.properties.get(GithubBasePluginConvention.GITHUB_USER_NAME_OPTION)?.toString()
         }))
 
         extension.password.set(project.provider({
-            def value = project.properties.get(GithubBasePluginConvention.GITHUB_USER_PASSWORD_OPTION)
-            if (value) {
-                return value.toString()
-            }
-            null
+            project.properties.get(GithubBasePluginConvention.GITHUB_USER_PASSWORD_OPTION)?.toString()
```

Comment:
I think this will not return the same or?
The idea is to return the property which from the API is an object (hence the to String)
But if the value is not set we want to return `null`. Will this code return `"null"` now? Or will the `?` shortcircut?

Reply:
it will return the same. The '?' operator makes the expression return null if its left side is null. 

`a?.b` is a shorthand for `if(a){ a.b } else{ null }`

Reply:
Also, I believe that this code is tested, so if anything went wrong, that should had fired up.

Reply:
Here explains it in a very consise manner: https://groovy-lang.org/operators.html#_safe_navigation_operator

Reply:
Ok fine by me.